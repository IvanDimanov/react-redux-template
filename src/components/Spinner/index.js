import React, {memo} from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'

import spinner from 'assets/images/spinner.svg'

import styles from './index.module.scss'


const Spinner = ({full, alt, className, size}) => (
  <img
    alt={alt}
    src={spinner}
    width={size}
    className={clsx(className, full && styles.Full)}
  />
)


Spinner.propTypes = {
  full: PropTypes.bool,
  alt: PropTypes.string,
  className: PropTypes.string,
  size: PropTypes.number,
}

Spinner.defaultProps = {
  full: false,
  alt: 'Loading ...',
  className: '',
  size: 150,
}

Spinner.displayName = 'Spinner'


export default memo(Spinner)
