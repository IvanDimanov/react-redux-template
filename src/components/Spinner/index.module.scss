$baseDimension: 200px;

.Full {
  position: absolute;
  top: calc(50% - #{$baseDimension / 2});
  left: calc(50% - #{$baseDimension / 2});
}
