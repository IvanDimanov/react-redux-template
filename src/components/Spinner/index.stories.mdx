import {Meta, Preview, Story, Props} from '@storybook/addon-docs/blocks';
import {withKnobs, boolean, number} from '@storybook/addon-knobs';

import Spinner from '../Spinner';

<Meta
  title='Design System/Components/Spinner'
  component={Spinner}
  decorators={[withKnobs]}
/>


# Component/Spinner
A basic UI Spinner component that allows to indicate a loading state.


#### The Default view renders simple Spinner.
<Preview>
  <Story name='Default'>
    <Spinner/>
  </Story>
</Preview>


## Spinner Size
Spinner with different size.
<Preview>
  <Story name='Spinner Size'>
    <Spinner size={number('size', 50)} />
  </Story>
</Preview>
