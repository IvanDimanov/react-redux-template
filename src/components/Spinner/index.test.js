import React from 'react'

import {shallow} from 'enzyme'

import Spinner from './index'

describe('Spinner Component', () => {
  describe('Basic Rendering', () => {
    let wrapper
    beforeEach(() => {
      wrapper = shallow(<Spinner />)
    })

    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
