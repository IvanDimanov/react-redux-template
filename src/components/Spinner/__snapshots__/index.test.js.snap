// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component Basic Rendering should match the snapshot 1`] = `"<img alt=\\"Loading ...\\" src=\\"spinner.svg\\" width=\\"150\\" class=\\"\\"/>"`;
